ReleaseProcessAutomation/Steps/PipelineSteps/ReleasePatchStep.cs

```diff
@@ -50,34 +51,37 @@ public class ReleasePatchStep : ReleaseProcessStepBase, IReleasePatchStep
       IInputReader inputReader,
       IMSBuildCallAndCommit msBuildCallAndCommit,
       IContinueReleasePatchStep continueReleasePatchStep,
+      IPushNewReleaseBranch pushNewReleaseBranch,
       IAnsiConsole console)
       : base(gitClient, config, inputReader, console)
   {
     _msBuildCallAndCommit = msBuildCallAndCommit;
     _continueReleasePatchStep = continueReleasePatchStep;
+    _pushNewReleaseBranch = pushNewReleaseBranch;
   }
 
   public void Execute (SemanticVersion nextVersion, string? commitHash, bool startReleasePhase, bool pauseForCommit, bool noPush, bool onMaster)
   {
     EnsureWorkingDirectoryClean();
 
-    if (GitClient.GetCurrentBranchName() == null)
+    var currentBranchName = GitClient.GetCurrentBranchName();
+    if (currentBranchName == null)
     {
       const string message = "Could not get the name of the current branch.";
       throw new InvalidOperationException(message);
     }
 
     if (onMaster)
     {
-      if (!GitClient.IsOnBranch("master"))
+      if (!currentBranchName.StartsWith("master"))
       {
         const string message = "Cannot release a patch for master when not on master branch.";
         throw new InvalidOperationException(message);
       }
     }
     else
     {
-      if (!GitClient.IsOnBranch("hotfix/"))
+      if (!currentBranchName.StartsWith("hotfix/"))
```

Comment:
Do we need those changes? They seem irrelevant to the issue at hand.

Reply:
Well we need the current branch name already, so why not use it and save some git calls? I know this doesnt fit this issue, perfectly, it does come in contact with it however. 

Can still revert it if that is not enough of an argument ;)

Reply:
`GitClient.IsOnBranch("bla")` and `currentBranch.StartsWith("bla")` do not seem like they test the same condition. We can keep the changes but we should address that issue.

Reply:
Not quite sure how to address that issue, naming? Wrapper method? Git branches are defined by names in this implementation, so I think it should seem like startsWith checks the same, but IDK.